File: MPP_MusicFestival_Java/FestivalServices/src/main/java/services/IFestivalService.java
package services;

import domain.OfficeEmployee;
import domain.Show;

import java.sql.Timestamp;
import java.util.List;

public interface IFestivalService {

    List<Show> getShows() throws LogException;

    List<Show> getArtistsByDate(Timestamp date) throws LogException;

    void buyTicket(Long showId, Long officeEmployeeId, String buyerName) throws LogException;

    void updateShow(Long showId, Show newShow) throws LogException;

    Show findShowById(Long showId) throws LogException;

    OfficeEmployee findEmployeeByUsername(String username) throws LogException;


    void logIn(OfficeEmployee officeEmployee, IFestivalObserver client) throws LogException;

    void logOut(OfficeEmployee officeEmployee, IFestivalObserver client) throws LogException;

}

File: services/IFestivalService.cs
using MusicFestival.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicFestival.services
{
    public interface IFestivalService
    {
        List<Show> getShows();

        List<Show> getArtistsByDate(DateTime date);

        void buyTicket(long showId, long officeEmployeeId, string buyerName);

        void updateShow(long showId, Show newShow);

        Show findShowById(long showId);

        OfficeEmployee findEmployeeByUsername(string username);


        void LogIn(OfficeEmployee officeEmployee, IFestivalObserver client);

        void LogOut(OfficeEmployee officeEmployee, IFestivalObserver client);
    }
}
